ably/realtime/connection.py

```diff
@@ -131,13 +132,31 @@ def __init__(self, realtime, initial_state):
         self.__ping_future = None
         self.__timeout_in_secs = self.options.realtime_request_timeout / 1000
         self.transport: WebSocketTransport | None = None
+        self.__ttl_task = None
+        self.__retry_task = None
+        self.__connection_details = None
+        self.__in_suspended_state = False
         super().__init__()
 
     def enact_state_change(self, state, reason=None):
         current_state = self.__state
         self.__state = state
+        if self.__state == ConnectionState.DISCONNECTED:
+            if not self.__ttl_task or self.__ttl_task.done():
+                self.__ttl_task = asyncio.create_task(self.__connection_state_ttl())
         self._emit('connectionstate', ConnectionStateChange(current_state, state, reason))
 
+    async def __connection_state_ttl(self):
```

Comment:
This method name is a bit confusing to me, could you rename it to something like `start_suspended_timer`?